spring-cloud-netflix-hystrix/src/main/java/org/springframework/cloud/netflix/hystrix/security/HystrixSecurityAutoConfiguration.java

```diff
@@ -56,18 +57,36 @@ public void init() {
 				.getPropertiesStrategy();
 		HystrixCommandExecutionHook commandExecutionHook = HystrixPlugins.getInstance()
 				.getCommandExecutionHook();
+		HystrixConcurrencyStrategy concurrencyStrategy = getConcurrencyStrategy();
 
 		HystrixPlugins.reset();
 
 		// Registers existing plugins excepts the Concurrent Strategy plugin.
 		HystrixPlugins.getInstance().registerConcurrencyStrategy(
-				new SecurityContextConcurrencyStrategy(existingConcurrencyStrategy));
+				new SecurityContextConcurrencyStrategy(concurrencyStrategy));
 		HystrixPlugins.getInstance().registerEventNotifier(eventNotifier);
 		HystrixPlugins.getInstance().registerMetricsPublisher(metricsPublisher);
 		HystrixPlugins.getInstance().registerPropertiesStrategy(propertiesStrategy);
 		HystrixPlugins.getInstance().registerCommandExecutionHook(commandExecutionHook);
 	}
 
+	private HystrixConcurrencyStrategy getConcurrencyStrategy() {
+		HystrixConcurrencyStrategy registeredStrategy = HystrixPlugins.getInstance()
```

Comment:
The logic here is kind of confusing the way it is written.  For example it seems that if `existingConcurrencyStrategy` is not `null` we could just return that right away.  I also dont think that the `IllegalStateException` is necessary, we should just default to the `existingConcurencyStrategy` if there is both registered (my suggestion above would eliminate the need for this).

Reply:
> For example it seems that if `existingConcurrencyStrategy` is not `null` we could just return that right away

Good idea. will `detectRegisteredConcurrencyStrategy()` be better ?



> I also dont think that the `IllegalStateException` is necessary, we should just default to the `existingConcurencyStrategy` if there is both registered

If both registered and they are not same, we really don‘t know which to select. No matter which one we choose, we lose the logic in another one. For `IllegalStateException`, this is the some style with Hystrix. (In Hystrix when register a `ConcurrencyStrategy` if there already has one, IllegalStateException was thrown)